src/extension/cell.ts

```diff
@@ -126,7 +134,7 @@ export class NotebookCellOutputManager {
       case OutputType.vercel: {
         const json: CellOutputPayload<OutputType.vercel> = {
           type: OutputType.vercel,
-          output: metadata['runme.dev/vercelState'] ?? {
+          output: this.cellState?.state as VercelState ?? {
```

Comment:
Implementation of `ICellState` is fine, except for one thing:

If the type of a cell changes, say, from `GithubState` to `DenoState`, then the existing state object could get carried over, resulting in the deno object being passed nonsense state.

We could fix this as follows: instead of accessing `this.cellState` directly, we use a `this.getCellState(type: OutputType)` method. When the existing cell state is not of the given type, the cell state could be reset.

Reply:
Thanks @mxsdev, added a new method as recommended